includes/libs/rdbms/database/Database.php

```diff
@@ -1205,18 +1205,28 @@ public function query( $sql, $fname = __METHOD__, $flags = 0 ) {
 		$this->lastQuery = $sql;
 
 		if ( $this->isWriteQuery( $sql ) ) {
-			# In theory, non-persistent writes are allowed in read-only mode, but due to things
-			# like https://bugs.mysql.com/bug.php?id=33669 that might not work anyway...
-			$this->assertIsWritableMaster();
+			/**
+			 * Fandom change - start (@author ttomalak)
+			 * Apply @mszabo patch https://gerrit.wikimedia.org/r/c/mediawiki/core/+/617741 to
+			 * allow Temporary table creation on DB_REPLICA for eg. SMW
+			 * PLATFORM-4964
+			 */
 			# Do not treat temporary table writes as "meaningful writes" that need committing.
 			# Profile them as reads. Integration tests can override this behavior via $flags.
 			$pseudoPermanent = $this->hasFlags( $flags, self::QUERY_PSEUDO_PERMANENT );
 			$tableType = $this->registerTempTableWrite( $sql, $pseudoPermanent );
 			$isEffectiveWrite = ( $tableType !== self::TEMP_NORMAL );
 			# DBConnRef uses QUERY_REPLICA_ROLE to enforce the replica role for raw SQL queries
-			if ( $isEffectiveWrite && $this->hasFlags( $flags, self::QUERY_REPLICA_ROLE ) ) {
-				throw new DBReadOnlyRoleError( $this, "Cannot write; target role is DB_REPLICA" );
+			if ( $isEffectiveWrite ) {
+				# In theory, non-persistent writes are allowed in read-only mode, but due to things
+				# like https://bugs.mysql.com/bug.php?id=33669 that might not work anyway...
```

Comment:
We can remove this comment as this is misleading/outdated (the referenced MySQL bug was closed over 10 years ago). I removed it in the upstream patch as well.

Reply:
👍  Removed